AWSIoT/AWSIoTDataManager.m

```diff
@@ -912,22 +912,39 @@ - (void)createSubscriptionsForShadow:(AWSIoTDataShadow *)shadow
  */
 - (BOOL)handleSubscriptionsForShadow:(NSString *)name
                             callback:(AWSIoTMQTTExtendedNewMessageBlock)callback {
+    return [self handleSubscriptionsForShadow:name callback:callback completionHandler:nil];
+}
+
+- (BOOL)handleSubscriptionsForShadow:(NSString *)name
+                            callback:(AWSIoTMQTTExtendedNewMessageBlock)callback
+                  completionHandler:(void(^)(void))completionHandler {
     BOOL rc = NO;
     AWSIoTDataShadow *shadow = (AWSIoTDataShadow *)[self.shadows objectForKey:name];
     
     if (shadow != nil) {
+        __block NSUInteger remainingTopics = [shadow.topics count];
```

Comment:
is `__block` used so that the changes to `remainingTopics` are visible inside this block when `ackCallback` is called?

Reply:
Exactly :) Just Objective-C stuff